pkgs/development/tools/mysql-shell/8.nix

```
@@ -34,41 +33,49 @@
 
 let
   pythonDeps = with python3.pkgs; [ certifi paramiko pyyaml ];
+
+  mysqlShellVersion = "8.4.1";
+  mysqlServerVersion = "8.4.1";
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "mysql-shell";
-  version = "8.0.37";
+  version = mysqlShellVersion;
 
   srcs = [
     (fetchurl {
-      url = "https://dev.mysql.com/get/Downloads/MySQL-${lib.versions.majorMinor finalAttrs.version}/mysql-${finalAttrs.version}.tar.gz";
-      hash = "sha256-4GOgkazZ7EC7BfLATfZPiZan5OJuiDu2UChJ1fa0pho=";
+      url = "https://dev.mysql.com/get/Downloads/MySQL-${lib.versions.majorMinor mysqlServerVersion}/mysql-${mysqlServerVersion}.tar.gz";
+      hash = "sha256-20Hxl9cXDFTX7cDQyaJzDCJfSvBeztD2S+z5u2wRAT4=";
     })
     (fetchurl {
       url = "https://dev.mysql.com/get/Downloads/MySQL-Shell/mysql-shell-${finalAttrs.version}-src.tar.gz";
-      hash = "sha256-UtZ7/Ip5h9CXKy3lkSt8/TXJgbPPUO73rMSIFPfX0Is=";
+      hash = "sha256-wTwoaoaCrTQqaqgE35Sg8zn8HqjsjQowbtgMZTpkYQU=";
     })
   ];
 
   sourceRoot = "mysql-shell-${finalAttrs.version}-src";
 
   postUnpack = ''
-    mv mysql-${finalAttrs.version} mysql
+    mv mysql-${mysqlServerVersion} mysql
   '';
 
+  patches = lib.optionals stdenv.isDarwin [
```

```suggestion
  patches = [
```
we want to notice when the patches no longer apply when updating

---

In fact, I am not very sure either; it seems that this change was intentionally made by the MySQL Shell Team. It might be a permanent modification. Obviously, it will affect Nix on Darwin.